utils/src/test/java/software/amazon/awssdk/utils/SdkHttpUtilsTest.java

```diff
@@ -173,4 +178,16 @@ public void headersFromCollectionWorksCorrectly() {
         assertThat(SdkHttpUtils.firstMatchingHeaderFromCollection(headers, asList("foo", "nothing"))).hasValue("bar");
         assertThat(SdkHttpUtils.firstMatchingHeaderFromCollection(headers, asList("foo", "other"))).hasValue("foo");
     }
+
+    @Test
+    public void uriParams() throws URISyntaxException {
+        URI uri = new URI("https://github.com/aws/aws-sdk-java-v2/issues/2034?reqParam=1234&oParam=3456&reqParam=5678&noval"
+                          + "&decoded%26Part=equals%3Dval");
+        Map<String, List<String>> uriParams = SdkHttpUtils.uriParams(uri);
+        assertThat(uriParams).contains(entry("reqParam", Arrays.asList("1234", "5678")),
+                                       entry("oParam", Collections.singletonList("3456")),
+                                       entry("noval", Arrays.asList("")),
```

Comment:
As mentioned above, the value for `noval` should be null, not a list containing `""`.